common/app/Panes/redux/index.js

```diff
@@ -176,14 +171,22 @@ export default function createPanesAspects({ createPanesMap }) {
           ...state,
           panesMap,
           panes,
-          panesByName: panes.reduce((panes, { name }, index) => {
+          panesByName: panes.reduce((__panes, { name }, index) => {
             const dividerLeft = utils.getDividerLeft(numOfPanes, index);
-            panes[name] = {
+            __panes[name] = {
               name,
-              dividerLeft,
-              isHidden: name === 'Map' ? state.isMapPaneHidden : false
+              dividerLeft: _.get(
+                state,
+                ['panesByName', name, 'dividerLeft'],
+                dividerLeft
+              ),
```

Comment:
instead of this, perhaps I could add this before `panesMap` is defined (above line 167): 
```js
const wasStepChallenge = _.findIndex(state.panesMap, {name: ‘Step’}) !== -1;
```
this, right after `panesMap` is defined (line 168):
```js
const isStepChallenge = _.findIndex(panesMap, {name: ‘Step’}) !== -1;
```
and then replace lines 176-182 above with:
```js
dividerLeft: name === ‘Map’ && wasStepChallenge && !isStepChallenge 
  ? 25 
  : _.get(
    state,
    ['panesByName', name, 'dividerLeft'],
    dividerLeft
 ),
```
This should prevent "Map"'s `dividerLeft` from defaulting to `50` and covering the side panel when moving from a Map/Step challenge/intro. Haven't tested this out yet though.

Reply:
This approach tightly couples Panes to the content. We should avoid this.

What about iterating through the old and new panes left to right and assigning the previous value of dividerLeft to the new pane. Then, once a pane is detected that did not exist before, we split the remaining area to the remaining panes.

Reply:
@BerkeleyTrue Yes, that could work, but the result would be less balanced visually, because when moving from Step/Map to 4 panes, we'd have Map taking up 50% of the view width and the remaining panes ~16% each. Let me think about this, I might have another solution that wouldn't require any addition utils to iterate over the panes and would still keep the panes and content decoupled.

Reply:
Yeah that could be a problem. Let me know what you come up with.